### ISSUE TITLE
Better handling of start-up errors

### ISSUE THREAD

**Issue by dimateos** [phil294/GitLG#93]
There are a bunch of errors when you open the view before vscode source control has loaded any git repo. 
Or when you open (restoring last selected repo?) and the selected repo is not loaded yet.
This loading process can be fast with a few repos as in the following pic, but for large workspaces it can be quite slow.

![image](https://github.com/phil294/git-log--graph/assets/26868611/edf8038f-e9f9-468e-9fe8-a443be21085e)

Also happens a lot when the extension editor is open when you close vscode and then later restore the workspace along with all the editors (this is when it always occurs to me, and quite often)

<details><summary> vscode error bubbles </summary>

![image](https://github.com/phil294/git-log--graph/assets/26868611/0b21da7e-bca9-4776-a08b-83f2f26a0399)
</details>

<details><summary> console log </summary>

* Stiched together as idk of a better method to share it, feel free to suggest other way or request more info.
* Some lines might not be related to the extension, I capture where related messages started till the end:

![2024-04-17_(092202)](https://github.com/phil294/git-log--graph/assets/26868611/0eb86cc9-b5a0-4854-8431-6010485229c9)
</details>

> Dunno if related to https://github.com/phil294/git-log--graph/issues/82, but keep happening to me as of the last version. ATM I use the hotkey to open the view, but would be nice to have button to open+switch to different repos.

## Feature request
Probably these error messages are useful for unexpected errors during usage, but there could be a way to silent them when clearly the issue is that the source control repos are not loaded yet. Or maybe directly avoid doing the queries when so.

Basically Ive never seen them pop up other than in the explained situations.

Best regards!

**Comment by phil294** [comment]
I've never had these errors come up, but I guess it should be rather easy to reproduce, particularly given how much information you provided. And personally I'm a fan of big aggressive ugly error message popups as they are more likely to be reported than silent console.logs - as long as they're fixed rapidly

**Comment by dimateos** [comment]
It seems that the error bubbles no longer appear. But the errors when attempting to load a graph view of a repo yet to be detected by vscode git remains:

* Close a worskpace with the graph open and active. Then, when you repoen the same worskpace and all tabs etc are restored, the graph is empty as the vscode source control engine has not parsed the repos yet.

<details><summary> vscode console errors </summary>

![image](https://github.com/user-attachments/assets/028f4e52-8fa3-4052-a853-c0741e19eb4d)
![image](https://github.com/user-attachments/assets/3f2dafa7-8f3d-438b-9686-79bcdfe41221)

</details>

*** I actually also got errors bubbles too

![image](https://github.com/user-attachments/assets/da20b8ea-8127-4984-a304-44da939bf528)


---
A simple solution I suggest:

* Make the tab with the graph do not persist when closing the workspace (this is what mhutchie.git-graph extension does)
* For the graph in a view option, not so simple? probably some suggestions from the original post to silence the errors.
  * Could also try to reload after X seconds when the loading issue happens

Anyway this issue is not a deal breaker, just letting you know it still happens.

**Comment by phil294** [comment]
Yes, since the repos come in one after another, the saved-as-selected one might not be present, resulting in a non-resolvable conflict between index and actual repo path. This means that you'll see one repo selected above but the data below for another one. Fixing this requires substantial changes to the repo management. I believe the Git extension (which GitLG currently depends on) doesn't emit an "all repos have been loaded" event unfortunately, either. I'll have a look soon I can't really estimate this yet.

**Comment by dimateos** [comment]
> Make the tab with the graph do not persist when closing the workspace (this is what mhutchie.git-graph extension does)

Maybe this workaround is simple to add and good enough for the time being

**Comment by phil294** [comment]
repos are now loaded one after another and loading states updated accordingly
